### PR TITLE
fix issues with s390x CI on master (xdiff and others)

### PR DESCRIPTION
**Pull request by bfredl**
no code yet, but tracks the following error on master s390x CI:
```
FAILED   4 tests, listed below:
FAILED   test/functional/lua/xdiff_spec.lua @ 23: xdiff bindings can diff text w
ith no callback
test/functional/lua/xdiff_spec.lua:24: Expected objects to be the same.
Passed in:
(string) ''
Expected:
(string) '@@ -1 +1 @@
-Hello
+Helli
'

stack traceback:
        (tail call): ?
        test/functional/lua/xdiff_spec.lua:24: in function <test/functional/lua/
xdiff_spec.lua:23>

FAILED   test/functional/lua/xdiff_spec.lua @ 49: xdiff bindings can diff text w
ith callback
test/functional/lua/xdiff_spec.lua:54: Expected objects to be the same.
Passed in:
(table: 0x40009bb2b0) { }
Expected:
(table: 0x40009baaf0) {
 *[1] = {
    [1] = 1
    [2] = 1
    [3] = 1
    [4] = 1 } }

stack traceback:
        (tail call): ?
        test/functional/lua/xdiff_spec.lua:54: in function <test/functional/lua/
xdiff_spec.lua:49>

FAILED   test/functional/lua/xdiff_spec.lua @ 96: xdiff bindings can diff text w
ith hunk_lines
test/functional/lua/xdiff_spec.lua:97: Expected objects to be the same.
Passed in:
(table: 0x40009d9760) { }
Expected:
(table: 0x40009d9020) {
 *[1] = {
    [1] = 1
    [2] = 1
    [3] = 1
    [4] = 1 } }

stack traceback:
        (tail call): ?
        test/functional/lua/xdiff_spec.lua:97: in function <test/functional/lua/
xdiff_spec.lua:96>

FAILED   test/functional/lua/xdiff_spec.lua @ 105: xdiff bindings can diff text 
can run different algorithms
test/functional/lua/xdiff_spec.lua:129: Expected objects to be the same.
Passed in:
(string) ''
Expected:
(string) '@@ -1,4 +0,0 @@
-.foo1 {
-    margin: 0;
-}
-
@@ -7,0 +4,5 @@
+
+.foo1 {
+    margin: 0;
+    color: green;
+}
'

stack traceback:
        (tail call): ?
        test/functional/lua/xdiff_spec.lua:129: in function <test/functional/lua
/xdiff_spec.lua:105>

ERROR    1 error, listed below:
ERROR    test/functional/lua/xdiff_spec.lua @ 83: xdiff bindings can diff text w
ith error callback
test/helpers.lua:244: expected failure, but got success

stack traceback:
        test/helpers.lua:244: in function 'pcall_err_withfile'
        test/helpers.lua:250: in function 'pcall_err_withtrace'
        test/helpers.lua:259: in function 'pcall_err'
        test/functional/lua/xdiff_spec.lua:92: in function <test/functional/lua/
xdiff_spec.lua:83>


 4 FAILED TESTS
 1 ERROR
```

### CODE REVIEW

**Comment by bfredl** [comment]
Other know, consistent failures:
```
test/functional/api/vim_spec.lua:3238: Expected objects to be the same.
Passed in:
(table: 0x401a707450) {
  ...
 *[default] = false
  ...
}
Expected:
(table: 0x401a706b80) {
  ...
 *[default] = true
  ...
}

FAILED   test/functional/core/job_spec.lua @ 1007: jobs jobstop() kills entire process tree #6530
test/functional/core/job_spec.lua:1094: retry() attempts: 67
test/functional/core/job_spec.lua:1096: Expected objects to be the same.
Passed in:
(table: 0x4020e8eeb0) {
  [name] = 'sleep <defunct>'
  [pid] = 30035
  [ppid] = 1 }
Expected:
(userdata) 'vim.NIL'

stack traceback:
	test/helpers.lua:97: in function 'retry'
	test/functional/core/job_spec.lua:1094: in function <test/functional/core/job_spec.lua:1007>

FAILED   test/functional/lua/thread_spec.lua @ 203: thread vim.* diff
test/functional/lua/thread_spec.lua:215: Expected objects to be the same.
Passed in:
(table: 0x40034ad5d0) {
  [1] = 'notification'
  [2] = 'result'
 *[3] = {
   *[1] = '' } }
Expected:
(table: 0x4003582e90) {
  [1] = 'notification'
  [2] = 'result'
 *[3] = {
   *[1] = '@@ -1 +1 @@
-Hello
+Helli
' } }

stack traceback:
	(tail call): ?
	test/functional/lua/thread_spec.lua:215: in function <test/functional/lua/thread_spec.lua:203>

FAILED   test/functional/lua/thread_spec.lua @ 366: threadpool vim.* work
test/functional/lua/thread_spec.lua:378: Expected objects to be the same.
Passed in:
(table: 0x4003416710) {
  [1] = 'notification'
  [2] = 'result'
 *[3] = {
   *[1] = '' } }
Expected:
(table: 0x40041ac290) {
  [1] = 'notification'
  [2] = 'result'
 *[3] = {
   *[1] = '@@ -1 +1 @@
-Hello
+Helli
' } }

stack traceback:
	(tail call): ?
	test/functional/lua/thread_spec.lua:378: in function <test/functional/lua/thread_spec.lua:366>


RROR    test/functional/lua/ui_spec.lua @ 139: vim.ui open() validation
test/helpers.lua:144: Pattern does not match.
Pattern:
vim.ui.open: command failed %(%d%): { "[^"]+", .*"non%-existent%-file" }
Actual:
vim.ui.open: no handler found (tried: wslview, xdg-open)

stack traceback:
	test/helpers.lua:144: in function 'matches'
	test/functional/lua/ui_spec.lua:144: in function <test/functional/lua/ui_spec.lua:139>

ERROR    test/functional/lua/xdiff_spec.lua @ 83: xdiff bindings can diff text with error callback
test/helpers.lua:244: expected failure, but got success

stack traceback:
	test/helpers.lua:244: in function 'pcall_err_withfile'
	test/helpers.lua:250: in function 'pcall_err_withtrace'
	test/helpers.lua:259: in function 'pcall_err'
	test/functional/lua/xdiff_spec.lua:92: in function <test/functional/lua/xdiff_spec.lua:83>

ERROR    test/functional/plugin/man_spec.lua @ 198: :Man reports non-existent man pages for absolute paths
test/helpers.lua:144: Pattern does not match.
Pattern:
Error detected while processing command line:
man.lua: "no manual entry for /github/workspace/neovim/build/Xtest_tmpdir/T4054%.126"
Actual:
Error detected while processing command line:
/github/workspace/neovim/runtime/lua/man.lua:716: bad argument #1 to 'format' (string expected, got nil)

stack traceback:
	test/helpers.lua:144: in function 'matches'
	test/functional/plugin/man_spec.lua:205: in function <test/functional/plugin/man_spec.lua:198>I
```

Errors in `thread_spec.lua` are likely just about xdiff, not threads (as xdiff itself is failing).
In total 8+3 known expected failures.

**Comment by dundargoc** [comment]
Piggybacking on this PR in case you agree/have other suggestions: we should probably use `-D CI_BUILD=ON` to prevent the DBG nvimlog dumpster fire at the end. The problem is that there are a bunch of conversion errors that I believe are incorrect. My initial idea was to disable conversion warnings on s390x, but I noticed I get the same errors on amd64 with gcc 8 so I think it might be a problem with older gcc versions.

So my idea is to disable conversion warnings with gcc altogether, or for a all versions smaller than 11 (assuming my theory is correct).

File: .github/workflows/optional.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
             bash -c
             "
             apt-get -y update &&
-            DEBIAN_FRONTEND=noninteractive apt-get -y install build-essential cmake curl gettext ninja-build locales-all cpanminus git attr libattr1-dev  &&
+            time DEBIAN_FRONTEND=noninteractive apt-get -y install build-essential cmake curl gettext ninja-build locales-all cpanminus git attr libattr1-dev xdg-utils &&
```
Comment (dundargoc) [inline .github/workflows/optional.yml:37]: Just checking, is `time` supposed to be here?

**Comment by bfredl** [comment]
still 4 more failures, but with timeout change it is at least possible to run all functionaltests to completion.

**Pull request merged**